core/src/main/scala/chisel3/Bits.scala

```diff
@@ -148,6 +148,21 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends Element wi
 
   /** @group SourceInfoTransformMacro */
   final def do_extract(x: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = {
+    this.widthOption.foreach { thisWidth =>
+      if (thisWidth == 0) {
+        Builder.error(s"Cannot extract from zero-width")
+      } else if (thisWidth == 1) {
+        Builder.warning("Extracting from a one-width element")
```

Comment:
```suggestion
```
I don't think we can or should special case 1-bit, the concerns of https://github.com/chipsalliance/chisel3/issues/2946 aside, this would also ban zero-width indexes of 1-bit UInts which we absolutely need to allow because that easy to have in a generator.